src/frontend/contexts/DraftObservationContext.tsx

```diff
@@ -74,94 +76,123 @@ const MOVED_AWAY_THRESHOLD_METERS = 100;
  * update is 15m away. */
 const ACCURACY_MOVED_AWAY_FACTOR = 1.5;
 
-async function createDraftObservationLocationUpdator({
-  instance,
-  actions: {updatePosition},
-}: DraftObservationStore) {
-  let {granted: locationPermissionGranted} =
-    await Location.getForegroundPermissionsAsync();
-  let locationSubscriptionPromise: Promise<Location.LocationSubscription> | null =
-    null;
-  let appState: AppStateStatus = AppState.currentState;
-  let isNewlyCreatedDraftInStore = isNewlyCreatedDraft(instance.getState());
-
-  AppState.addEventListener('change', async nextAppState => {
-    appState = nextAppState;
-    if (appState === 'active' && !locationPermissionGranted) {
-      locationPermissionGranted = (
-        await Location.getForegroundPermissionsAsync()
-      ).granted;
-    }
-    watchPositionIfNeeded();
-  });
-
-  instance.subscribe(storeState => {
-    isNewlyCreatedDraftInStore = isNewlyCreatedDraft(storeState);
-    watchPositionIfNeeded();
-  });
-
-  async function watchPositionIfNeeded() {
-    const shouldBeWatchingPosition =
-      appState === 'active' &&
-      isNewlyCreatedDraftInStore &&
-      locationPermissionGranted;
-
-    if (shouldBeWatchingPosition && !locationSubscriptionPromise) {
-      locationSubscriptionPromise = Location.watchPositionAsync(
-        LOCATION_OPTIONS,
-        onPositionUpdate,
-      );
-    } else if (!shouldBeWatchingPosition && locationSubscriptionPromise) {
-      // Avoid a race condition by nulling the state before awaiting the promise
-      const locationSubscriptionPromiseCopy = locationSubscriptionPromise;
-      locationSubscriptionPromise = null;
-      const subscription = await locationSubscriptionPromiseCopy;
-      subscription.remove();
-    }
-  }
-
-  function onPositionUpdate(location: Location.LocationObject) {
-    const {value: currentDraft, initialPosition} = instance.getState();
-    if (!currentDraft) return;
-
-    const isManualLocation = !!currentDraft.metadata?.manualLocation;
-    if (isManualLocation) return;
-
-    const isStale =
-      Date.now() - location.timestamp > STALE_LOCATION_THRESHOLD_MS;
-    if (isStale) return;
-
-    instance.setState(prev => {
-      if (prev.initialPosition || !prev.value) return prev;
-      return {...prev, initialPosition: convertPosition(location)};
+function useCreateDraftObservationLocationUpdator(
```

Comment:
Updating this function might be out of scope. But this architecture does make updating the draft observation location easier and more straight forward to manage now.

Most notably, we can subscribe to location updates from the context while reading the location value from the Zustand store instead of React state. This means that when the location changes we can update the draft observation store location. And since we are using zustand store state, we do not need to worry about it causing unnecessary re-renders.